libsolidity/interface/StandardCompiler.cpp

```diff
@@ -787,6 +787,12 @@ std::variant<StandardCompiler::InputsAndSettings, Json::Value> StandardCompiler:
 		std::optional<langutil::EVMVersion> version = langutil::EVMVersion::fromString(settings["evmVersion"].asString());
 		if (!version)
 			return formatFatalError(Error::Type::JSONError, "Invalid EVM version requested.");
+		if (version <= EVMVersion::constantinople())
+			ret.errors.append(formatError(
+				Error::Type::Warning,
+				"general",
+				"Support for constantinople and older EVM versions is deprecated and will be removed in the future."
+			));
```

Comment:
I decided to put it in the interface (rather than compiler/yul stack) to avoid having to deal with filtering out the warning in soltest. The EVM version is a CLI argument so the warning would be present or missing in every test based on that.

The consequence is that it has no error code and that it won't appear when the use is not through CLI or Standard JSON (which I guess never happens for a normal user). The upside is that I don't have to track what uses the EVM version and instead I issue the warning at the point where the option gets its value.